tests/cpu_bugs/xen_mitigations.pm

```diff
@@ -0,0 +1,344 @@
+        },
+        "l1d-flush=on" => {
+            default => {
+                expected   => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+                unexpected => {''},
+            }
+        },
+    },
+  };
+
+sub do_check {
+    my $secnario = shift;
+    my $foo      = $secnario->{default};
+    if ($foo->{expected}) {
+        while (my ($cmd, $lines) = each %{$foo->{expected}}) {
+            foreach my $l (@{$lines}) {
+                if ($l ne "") {
+                    my $ret = script_run("$cmd | grep \"$l\"");
+                    record_info("ERROR", "Can't found a expected string.", result => 'fail') unless $ret eq 0;
+                } else {
+                    print "This expection is empty string, skip";
+                }
+
+            }
+        }
+    }
+    if ($foo->{unexpected}) {
+        while (my ($cmd, $lines) = each %{$foo->{unexpected}}) {
+            foreach my $l (@{$lines}) {
+                if ($l ne "") {
+                    my $ret = script_run("$cmd | grep \"$l\"");
+                    record_info("ERROR", "found a unexpected string.", result => 'fail') unless $ret ne 0;
+                } else {
+                    #Debug what output be report.
+                    assert_script_run("xl dmesg | grep -A 10 \"Speculative\"");
+                    print "This unexpection is empty string, skip";
+                }
+
+            }
+        }
+    }
+}
+
+sub do_test {
+    my ($self, $hash) = @_;
+    #xen parameter be store into arg
+    #
+    while (my ($arg, $dict) = each %$hash) {
+        while (my ($key, $value) = each %$dict) {
+            my $parameter = $arg . '=' . $key;
+            bootloader_setup::add_grub_xen_cmdline_settings($parameter);
+            bootloader_setup::grub_mkconfig();
+            Mitigation::reboot_and_wait($self, 150);
+            #$value include the check rules of current $parameter.
+            my $ret = do_check($value);
+            if ($ret ne 0) {
+                record_info('ERROR', "$parameter test is failed.", result => 'fail');
+            }
+            record_info('INFO', "$parameter test is finished.");
+            bootloader_setup::remove_grub_xen_cmdline_settings($parameter);
+            bootloader_setup::grub_mkconfig();
+        }
+
+    }
+}
+
+sub run {
+    my $self = @_;
+    select_console 'root-console';
+    my $ret = script_run("test -d /proc/xen");
+    if ($ret ne 0) {
+        record_info("platform mistake, This system is not running as Dom0.");
+        die;
+    }
+    do_test($self, $mitigations_list);
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    select_console 'root-console';
+    script_run(
+"md /tmp/upload_mitigations; cp " . $Mitigation::syspath . "* /tmp/upload_mitigations; cp /proc/cmdline /tmp/upload_mitigations; lscpu >/tmp/upload_mitigations/cpuinfo; tar -jcvf /tmp/upload_mitigations.tar.bz2 /tmp/upload_mitigations"
+    );
+    script_run("xl info");
```

Comment:
Temporary directories or files created on the system are best removed after the test is complete to restore the initial state of the system and avoid affecting subsequent tests.

Reply:
Agree!